### DISCUSSION TITLE
Xstate Actions not changing in persistent data

### DISCUSSION THREAD

**Discussion by vinodKumarPula** [statelyai/xstate#1937] [Q&A]
Hi, 
Im using Xstate in typescript on server side, here i have after delays, when machine is triggered and continous on it will work. when i want to restart it and when im making state changes the delays are not working. Please look and let me know if im doing wrong.

machine:
{
        "id" : "serviceMachine",
        "initial" : "ACTIVE",
        "context" : {
            "noTopUp" : "",
            "noUsage" : ""
        },
        "states" : {
            "ACTIVE" : {
                "after" : {
                    "NoTopUpToFree" : "checkTopup",
                    "NoUsage" : "SUSPEND"
                },
                "on" : {
                    "SUSPEND" : "SUSPEND",
                    "SYNC" : "ACTIVE",
                    "TERMINATE" : "TERMINATE"
                }
            },
            "checkTopup" : {
                "invoke" : {
                    "src" : "checkTopUpDB",
                    "onDone" : "ACTIVE",
                    "onError" : {
                        "target" : "SUSPEND"
                    }
                }
            },
            "SUSPEND" : {
                "after" : {
                    "ActiveDelay" : "ACTIVE"
                },
                "on" : {
                    "TERMINATE" : "TERMINATE",
                    "RESUME" : "ACTIVE"
                }
            },
            "TERMINATE" : {}
        }
    }

 let service=interpret( machine).onTransition(async (state: any) => {
})
  service.start(machine.resolveState(State.create(parsedObject))) 
  
this is how im interpreting machine

**Comment by vinodKumarPula** [comment]
delays: {
    NoTopUpToFree: (context: any, event: any) => {
      console.log('insdide notopup**********')
      return new Date(context.noTopUp).getTime() - Date.now()
    },
    NoUsage: (context: any) => {
      console.log('contex%%%%%%%%%%%%%%', context.noUsage)
      return new Date(context.noUsage).getTime() - Date.now()
    },
    ActiveDelay: (context: any) => {
      console.log('inside active Delay***********8', context.noUsage, new Date(context.noUsage).getTime() + 1000 - Date.now() > 0 ? new Date(context.noUsage).getTime() + 1000 - Date.now() : 1000)
      return new Date(context.noUsage).getTime() + 1000 - Date.now() > 0 ? new Date(context.noUsage).getTime() + 1000 - Date.now() : 1000
    }
    // NoUsage: () => {return 30 * (24 * 60 * 60 * 1000)}
  },
  services: {
    checkTopUpDB: (context: any, event: any) => {
      console.log('inside check topup')
      return new Promise((resolve, reject) => {
        setTimeout(() => {return reject('done')}, 2 * 1000)
      })
    }
  }

This are delays

**Reply by davidkpiano** [reply]
Please format your code in a readable format: https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks

Also, have you tried with normal delays (such as `1000`), just to test?
